UITests/TopSitesTests.swift

```diff
@@ -7,77 +7,47 @@ import WebKit
+        profile.prefs.setObject([], forKey: "topSites.deletedSuggestedSites")
         webRoot = SimplePageServer.start()
     }
 
-    func createNewTab(){
-        tester().tapViewWithAccessibilityLabel("Menu")
-        tester().tapViewWithAccessibilityLabel("New Tab")
-        tester().waitForViewWithAccessibilityLabel("Search or enter address")
-    }
-
-    func openURLForPageNumber(pageNo: Int) {
-        let url = "\(webRoot)/numberedPage.html?page=\(pageNo)"
-        tester().tapViewWithAccessibilityIdentifier("url")
-        tester().clearTextFromAndThenEnterTextIntoCurrentFirstResponder("\(url)\n")
-        tester().waitForWebViewElementWithAccessibilityLabel("Page \(pageNo)")
-    }
-
-    func rotateToLandscape() {
-        // Rotate to landscape.
-        let value = UIInterfaceOrientation.LandscapeLeft.rawValue
-        UIDevice.currentDevice().setValue(value, forKey: "orientation")
-    }
-
-    func rotateToPortrait() {
-        // Rotate to landscape.
-        let value = UIInterfaceOrientation.Portrait.rawValue
-        UIDevice.currentDevice().setValue(value, forKey: "orientation")
+    private func extractTextSizeFromThumbnail(thumbnail: ThumbnailCell) -> CGFloat? {
+        return thumbnail.textLabel.font.pointSize
     }
 
-    // A bit internal :/
-    func extractTextSizeFromThumbnail(thumbnail: UIView) -> CGFloat? {
-        guard let contentView = thumbnail.subviews.first, let wrapper = contentView.subviews.first,
-            let textWrapper = wrapper.subviews.last, let label = textWrapper.subviews.first as? UILabel else { return nil }
-
-        return label.font.pointSize
+    private func accessibilityLabelsForAllTopSites(collection: UICollectionView) -> [String] {
+        return collection.visibleCells().reduce([], combine: { arr, cell in
+            if let label = cell.accessibilityLabel {
+                return arr + [label]
+            }
+            return arr
+        })
     }
 
-    func testRotatingOnTopSites() {
-        // go to top sites. rotate to landscape, rotate back again. ensure it doesn't crash
-        createNewTab()
-        rotateToLandscape()
-        rotateToPortrait()
-
-        // go to top sites. rotate to landscape, switch to another tab, switch back to top sites, ensure it doesn't crash
-        rotateToLandscape()
-        tester().tapViewWithAccessibilityLabel("History")
-        tester().tapViewWithAccessibilityLabel("Top sites")
-        rotateToPortrait()
-
-        // go to web page. rotate to landscape. click URL Bar, rotate to portrait. ensure it doesn't crash.
-        openURLForPageNumber(1)
-        rotateToLandscape()
-        tester().tapViewWithAccessibilityIdentifier("url")
-        tester().waitForViewWithAccessibilityLabel("Top sites")
-        rotateToPortrait()
-        tester().tapViewWithAccessibilityLabel("Cancel")
+    // Quick way to clear out all our history items
+    private func clearPrivateDataFromHome() {
```

Comment:
Worth moving to `Global.swift`? Ideally, we'd clear history after *every* test to put the browser back into a clean of state as possible. If we did that, though, we'd probably want a way to clear even faster than using CPD (e.g., using notifications), so that's probably under the scope of [bug 1191476](https://bugzilla.mozilla.org/show_bug.cgi?id=1191476).